initial commit

Peter-Springer opened the pull request:
initial commit-questions and answers are rendered

<img width="1440" alt="screen shot 2016-11-06 at 9 53 01 pm" src="https://cloud.githubusercontent.com/assets/15174582/20046839/8174a756-a46b-11e6-9e58-bef2e21c3c1e.png">

@stevekinney
@brittanystoroz
@martensonbj

Peter-Springer committed:
initial commit

brittanystoroz commented on src/components/Quiz.js:
```diff
@@ -0,0 +1,54 @@
+  }
+
+  renderQuestionsToPage(state) {
+    return state.map(q => <article className='question-answers' key={q.id}>
```
Questions would be a good thing to break out into it's own component, so you can loop through all the questions and just render something like

```javascript
<Question answers={question.answers} title={question.title} key={question.id}
```

brittanydionigi replied:
We might want to break up this block of HTML we're returning by creating some additional components. If we add a `Question` component and an `Answer` component, we could render something like the following:

``` javascript
questions.map((question, index) =>
    <Question title={question.title} key={question.id}/>
    (question.answers).map(answer =>
        <Answer title={answer.title} />
    )
```

(That is not going to be the exact syntax, and I'm sure it will break in all sorts of ways, but that's what we should be going for here.)

brittanydionigi commented on src/components/Quiz.js:
```diff
@@ -0,0 +1,54 @@
+    super();
+    this.state = {
+      quizzes: [],
+      questions: [],
```
We probably don't need to save questions into our state, since we can access them directly through the quiz itself. A simpler state might just store the single quiz we get back when we hit our API endpoint.

``` javascript
this.state = {
    quiz: {}
}
```

Peter-Springer committed:
broke out components
arrow functions were not binding this correctly
binded this the old school way because of unexpected behavior from the arrow functions

Peter-Springer committed:
update styles to account for new components

Peter-Springer commented:
@brittanystoroz 
broke out components and refactored to use a single state

brittanydionigi commented on src/components/Questions.js:
```diff
@@ -0,0 +1,32 @@
+  }
+
+  loadQuestions() {
+    return this.props.questions.map(question => {
```
This isn't really solving the problem. The point of a component is to re-use the code that applies to multiple elements on the page. A `Questions` component isn't re-using anything. It's just moving the code from one file to another. There should be a `Question` component that represents each question, since all of the code for each question will be exactly the same. We shouldn't have to map over an array to render a component. 

Again, in your quiz component, we want to render something like:

``` javascript
questions.map((question, index) => {
    <Question title={question.title} key={question.id} />
    <div className="answer-group">
        (questions.answers).map((answer, index) => {
            <Answer title={answer.title} />
        })
    </div>
})
```

The Question component can be as simple as returning a paragraph tag that contains the title:

``` javascript
return (<p>{this.props.title}</p>)
```

And the answer component can be as simple as:

``` javascript
return (<label><input type="radio" name=`answer-group-{questionIndex}` />{answer.title}</label>
```

brittanydionigi reviewed (changes requested):
Needs some component refactoring.
